xarray/core/dataset.py

```diff
@@ -7074,5 +7074,43 @@ def query(
         # apply the selection
         return self.isel(indexers, missing_dims=missing_dims)
 
+    def drop_duplicates(
+        self,
+        dims: Union[None, Hashable, Sequence[Hashable]] = None,
+        keep: Union[str, bool] = "first",
+    ):
+        """Returns a new dataset with duplicate dimension values removed.
+
+        Parameters
+        ----------
+        dims : dimension label or sequence of labels, optional
+            Only consider certain dimensions for identifying duplicates, by
+            default use all of the columns.
+        keep : {"first", "last", False}, default: "first"
+            Determines which duplicates (if any) to keep.
+            - ``first`` : Drop duplicates except for the first occurrence.
+            - ``last`` : Drop duplicates except for the last occurrence.
+            - False : Drop all duplicates.
+
+        Returns
+        -------
+        Dataset
+        """
+        if dims is None:
+            dims = list(self.coords)
+        elif isinstance(dims, str) or not isinstance(dims, Iterable):
```

Comment:
You could in principle use `elif isinstance(dims, Hashable):` but I would leave it as is (we should once discuss what we do about `da.mean(("x", "y"))` as `("x", "y")` is `Hashable`)